dali/test/python/test_operator_get_property.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali import pipeline_def
+import os
+import numpy as np
+from nvidia.dali import fn
+from test_utils import get_dali_extra_path
+
+test_data_root = get_dali_extra_path()
+
+
+def _uint8_tensor_to_string(t):
+    return np.array(t).tobytes().decode()
+
+
+@pipeline_def
+def file_properties(root_path, device):
+    read, _ = fn.readers.file(file_root=root_path)
+    if device == 'gpu':
+        read = read.gpu()
+    return fn.get_property(read, key="source_info")
+
+
+def _test_file_properties(device):
+    root_path = os.path.join(test_data_root, 'db', 'single', 'png')
```

Comment:
nitpick: This is now making assumptions on what fn.readers.file will discover and in what order. Maybe it's more intuitive if we pass the file_list directly? The test would be more readable, I mean

Reply:
Good point. Done.